service/src/Helper.cpp

```diff
@@ -210,17 +210,27 @@ namespace geopm
     void check_hint(uint64_t hint)
     {
         if ((hint & ~GEOPM_MASK_REGION_HINT) != 0ULL) {
-            throw Exception("Helper::" + std::string(__func__) + "(): invalid hint: " +
+            throw Exception("Helper::" + std::string(__func__) +
+                            "(): invalid hint: " +
                             geopm::string_format_hex(hint),
                             GEOPM_ERROR_INVALID, __FILE__, __LINE__);
         }
-        if (hint && ((hint & (hint - 1)) != 0)) {   /// power of 2 check
-            throw Exception("Helper::" + std::string(__func__) + "(): multiple region hints set and only 1 at a time is supported: " +
+        unsigned int hint_idx = hint_to_index(hint);
+        if (hint_idx >= GEOPM_SENTINEL_REGION_HINT) {
+            throw Exception("Helper::" + std::string(__func__) +
+                            "(): hint out of range: " +
                             geopm::string_format_hex(hint),
                             GEOPM_ERROR_INVALID, __FILE__, __LINE__);
         }
     }
 
+    unsigned int hint_to_index(uint64_t hint)
```

Comment:
This is perhaps not the best file for this function, but placed it here for now. Wanted to get a draft PR out there for now. Will revisit this in the meantime.

Reply:
A bit of context: I decided to put it there for now to follow the current pattern, as I noticed another hint enum helper function was placed there as well (``check_hint()``).

However, thinking about it a bit more, I would propose these functions (``check_hint()``, ``hint_to_index()``) be moved to ``geopm_hint.h`` for cohesion. The functions in ``Helper.hpp`` are more general-purpose, while the hint functions are directly tied to the hint enum. Also, keeping them together with the enum will make them easier to maintain, and you can reduce header dependencies as code that needs to use the enum will only need to include one header (as opposed to two).

Reply:
sounds good!